### ISSUE TITLE
Should we make one type per node?

### ISSUE THREAD

**Issue by cbizon** [RobokopU24/Issues#97]
When we use the neo4j db analyzer to predict our page size and memory usage, the results produce a very unstable db.   One possible reason is that we are doing something odd by having many labels on each node.  And each of those labels has its own id index.  This might lead to some serious memory problems, but it enables biolink class entailment in queries, which is great.

But maybe we should explore the use only of the most specific type(s) on each node.   In that case, we'd need some preprocessing in the transpiler like we do for predicates and qualifiers in order to do the entailment properly.

**Comment by cbizon** [comment]
Re-evaluate post neo4j 5

**Comment by hina-shah** [comment]
Is there a status on this?

**Comment by EvanDietzMorris** [comment]
No, but we should keep this open. I think it's a very valid road to go down and I've been thinking about it again very recently. The inclusion of, and probably more importantly, indexing of, every node type in the biolink hierarchy is probably costing us, making the neo4j dbs larger, and eating up memory storing those indexes in memory, when most types are never queried for.
